languages/elixir/exercises/concept/list-comprehensions/.meta/example.ex

```diff
@@ -0,0 +1,15 @@
+defmodule BoutiqueSuggestions do
+  def get_combinations(tops, bottoms, options \\ []) do
+    maximum_price = Keyword.get(options, :maximum_price, 100.00)
+
+    for top <- tops,
+        %{base_color: top_base_color, price: top_price} = top,
+        bottom <- bottoms,
+        %{base_color: bottom_base_color, price: bottom_price} = bottom,
+        top_base_color != bottom_base_color,
+        (top_price + bottom_price) <= maximum_price
```

Comment:
As someone that doesn't use list comprehensions, I'll say: 🤯. But it's totally fine as long as the instructions are split into a few steps.

Reply:
I think from reading the getting started guide, I would realistically expect students to write the generator and a match on one line, but I broke it to two lines for clarity for what's happening in the example:

```
iex(4)> x = fn t, b -> 
...(4)>   for %{base_color: top_base_color, price: top_price} = top <- t,
...(4)>       %{base_color: bottom_base_color, price: bottom_price} = bottom <- b,
...(4)>       top_base_color != bottom_base_color,
...(4)>       (top_price + bottom_price) <= 100.00 do    
...(4)>     {top, bottom}
...(4)>   end
...(4)> end
```

Reply:
And yes, it will be over a few steps, I have ordered the tests to hopefully reflect the progression of the steps.